client/homebrew/homebrew.jsx

```diff
@@ -12,6 +13,21 @@ const NewPage = require('./pages/newPage/newPage.jsx');
 //const ErrorPage = require('./pages/errorPage/errorPage.jsx');
 const PrintPage = require('./pages/printPage/printPage.jsx');
 
+const WithRoute = (props)=>{ //(Element, additionalPropsFn)=>{
+	const params = useParams();
+	const location = useLocation();
+	let additionalProps = {};
+	if(props.additionalPropsFn) {
+		additionalProps = props.additionalPropsFn({ props, params, location });
+	}
+	const Element = props.el;
+	return <Element {...{
+		...props,
+		el                : undefined,
+		additionalPropsFn : undefined
+	}} {...params} {...additionalProps}/>;
```

Comment:
If we remove `additionalPropsFn` and use `queryParams` as I noted above, I think we can simplify this to (something like):

```suggestion
	const Element = props.el;
	props.el = undefined;
	return <Element {...props, ...params, ...queryParams}/>;
```

Reply:
The curly braces will have to be one set per variable we're destructuring, just as a result of how jsx handles it. 

I did originally try to modify `props` like what you have in your suggestion and React said no -- it's apparently immutable.